src/sinks/influxdb/logs.rs

```diff
@@ -97,6 +105,24 @@ pub struct InfluxDbLogsConfig {
         skip_serializing_if = "crate::serde::skip_serializing_if_default"
     )]
     acknowledgements: AcknowledgementsConfig,
+
+    /// Use this option to customize the key containing the hostname.
+    ///
+    /// The setting of `log_schema.host_key`, usually `host`, is used here by default.
+    #[configurable(metadata(docs::examples = "hostname"))]
+    pub host_key: Option<OptionalValuePath>,
```

Comment:
It looks like a case where we can remove the Option, and instead provide a serde default getting that from the log_schema.host_key.

Same goes for `source_type_key` and `message_key`.

Reply:
Well, it wouldn't trigger that check since it doesn't have a `serde(default)`, it just unwraps the `None` and handles it internally.

I don't like this pattern, but think it's a a temporary step as we transition across the log namespace changes. Not sure if we want to keep it permanently.

Reply:
Ah I mistook the example as a default my bad.

I do think it's worth considering a comment, simply because that consideration for the vector generate use case is not obvious from the code itself but it is very valid.

Reply:
Yeah, I'm not sure it's the best option or if it matches what's done elsewhere.... but 🤷

Reply:
Added in [9cdcee7](https://github.com/vectordotdev/vector/pull/16548/commits/9cdcee73bdd2ddcc05b2ff9b8bc2463258bb0ff3)